radis/misc/profiler.py

```diff
@@ -0,0 +1,74 @@
+"""
+`Profiler` class, for printing time spent on various steps during Spectrum
+calculation under :py:class:`~radis.lbl.factory.SpectrumFactory` based on verbose level.
+
+Also stores Spectrum calculation time dependent parameters, under the attribute
+    :py:attr:`~radis.misc.profiler.Profiler.dict_time`
+
+Routine Listing
+---------------
+
+- :meth:`~radis.misc.profiler.Profiler.start`
+- :meth:`~radis.misc.profiler.Profiler.stop`
+- :meth:`~radis.misc.profiler.Profiler._print`
+
+-------------------------------------------------------------------------------
+"""
+
+from time import time
+
+from radis.misc.printer import printg
+
+
+class Profiler(object):
+    """A class to store Spectrum calculation time dependent parameters, under the attribute
+    :py:attr:`~radis.misc.profiler.Profiler.dict_time` of
+    :py:class:`~radis.lbl.factory.SpectrumFactory`.
+
+    It also hold functions to print all the entities based on verbose value.
+
+    See Also
+    --------
+
+    :py:attr:`~radis.lbl.loader.DatabankLoader.input`,
+    :py:attr:`~radis.misc.profiler.Profiler,
+
+    """
+
+    def __init__(self, verbose):
+        super(Profiler, self).__init__()
+
+        # Dev: Init here to be found by autocomplete
+        self.initial = {}
+        self.dict_time = {}
+        self.verbose = verbose
+
+    def start(self, key, verbose):
+        if __debug__:
+            self.initial[key] = {
+                "start_time": time(),
+                "verbose": verbose,
+            }
+
+    def stop(self, key, details=""):
+        if __debug__:
+            items = self.initial.pop(key)
+            self.dict_time[key] = time() - items["start_time"]
+            if self.verbose >= items["verbose"]:
+                self._print(key, items["verbose"], details)
+
```

Comment:
Looks like you don't use self.verbose. Just a thought, but you could start with self.verbose = 0, and every time start() is called self.verbose is incremented, and when stop() is called self.verbose is decremented. This way you don't need to explicitly specify the verbosity when profiling.

Reply:
I'd call the inside key "level"  rather than verbose. And have "if... Verbose>=level then print ()"

Reply:
@anandxkumar if you can just update the name 
```
if self.verbose >= items["verbose"]
```
to 
```
 items["level"]     or items["verbose_level"]
```